### ISSUE TITLE
Fix issues raised by make lesson-check

### ISSUE THREAD

**Issue by fmichonneau** [carpentries/lesson-example#321]
```
 _episodes/03-organization.md:87:    look for uses of `FIXME`.
./_episodes/04-formatting.md:178: Unknown or missing code block type language-html
./_episodes/04-formatting.md:244: Unknown or missing code block type language-yaml
./_episodes/05-rmarkdown-example.md:104: Unknown or missing blockquote type None
./_episodes/05-rmarkdown-example.md:123: Unknown or missing blockquote type None
./_episodes/06-style-guide.md:198: Unknown or missing blockquote type None
./_episodes/06-style-guide.md:202: Unknown or missing blockquote type None
make: *** [lesson-check] Error 1
Makefile:135: recipe for target 'lesson-check' failed
```

**Comment by zkamvar** [comment]
I think some of these issues may be fixed by https://github.com/carpentries/styles/pull/532. I'll make a PR to check.

**Comment by zkamvar** [comment]
Updating styles gives us:

```
_episodes/03-organization.md:87:    look for uses of `FIXME`.
./404.md:11: Unknown or missing blockquote type caution
./_episodes/02-tooling.md:124: Unknown or missing code block type html
./_episodes/05-rmarkdown-example.md:104: Unknown or missing blockquote type None
./_episodes/05-rmarkdown-example.md:123: Unknown or missing blockquote type None
./_episodes/06-style-guide.md:109: Unknown or missing code block type html
./_episodes/06-style-guide.md:174: Unknown or missing code block type html
./_episodes/06-style-guide.md:198: Unknown or missing blockquote type None
./_episodes/06-style-guide.md:202: Unknown or missing blockquote type None
```

**Comment by vinisalazar** [comment]
Hi!

I was working on swcarpentry/python-novice-inflammation#498 and it was noted that when running `make lesson-check-all`, we had warnings for long lines in the `CONTRIBUTING.md` file (and the `README.md` file as well, but that is a different issue as noted in swcarpentry/python-novice-inflammation#911).

So, it would be nice to have this fixed in this repository so that future lessons do not have the same warnings.

Running `make lesson-check-all` in this repository, I got the following output:
```
./CONTRIBUTING.md: Line(s) too long: 3, 114, 126, 131, 135
./README.md: Line(s) too long: 1
./_episodes/02-tooling.md:124: Unknown or missing code block type html
./_episodes/03-organization.md: Line(s) too long: 92, 162
./_episodes/04-formatting.md: Line(s) too long: 112, 113, 115
./_episodes/05-rmarkdown-example.md: Line(s) too long: 42, 145, 162
./_episodes/06-style-guide.md: Line(s) too long: 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 188, 200, 204
./_episodes/06-style-guide.md:109: Unknown or missing code block type html
./_episodes/06-style-guide.md:174: Unknown or missing code block type html
./_episodes/06-style-guide.md:198: Unknown or missing blockquote type None
./_episodes/06-style-guide.md:202: Unknown or missing blockquote type None
./_episodes/07-checking.md: Line(s) too long: 55
./index.md: Line(s) too long: 9, 12, 33
./setup.md: Line(s) too long: 30, 96, 212, 236, 274, 275, 278, 443, 531
```

I noticed that the difference between `make lesson-check` and `make lesson-check-all` is that the latter adds the `-l`, `-w`, `--permissive` flags to the `lesson_check.py` command. So these warnings would be overlooked from running `make lesson-check`.

I will submit a PR to try to fix some of these line length warnings, and see if I can also fix some of the code block ones.

Best,

V

**Comment by vinisalazar** [comment]
Hi, I just submitted a PR for this.

I have a few observations:
- the 'Unknown or missing code block type' warning happens because only code blocks declared in the `KNOWN_CODEBLOCKS` variable of `bin/lesson_check.py` are considered valid by the script. These code blocks are declared in the CSS styles at `assets/css/lessons.scss`. One could create styles for an `html` tag, but I thought it would be simpler to convert all `html` tags to `source` tags;
- the same goes for blockquotes: they must be declared in the `KNOWN_BLOCKQUOTES` variable of `lesson_check.py`. Since there are blockquotes which don't have any particular style (e.g. lines 198 and 202 of `_episodes/06-style-guide.md`, I simply added a `None` type to that variable. This way we can have these "plain" blockquotes without raising any warnings;
- some lines contain long URLs or HTML tags and thus it is impractical to contain them in the 100 char limit (e.g. L164 of `_episodes/05-rmarkdown-example.md` or L14 of `index.md`);
- I added a link to the Curriculum Development Handbook in `_includes/links.md`, as I thought it is a resource that may likely be referenced again in the future.

I hope it helps!

Best,
V
